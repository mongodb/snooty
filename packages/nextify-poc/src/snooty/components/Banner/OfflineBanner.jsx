import React from 'react';
import PropTypes from 'prop-types';
import LeafyBanner from '@leafygreen-ui/banner';
import { cx } from '@leafygreen-ui/emotion';
import Link from '../Link';
import { assertTrailingSlash } from '../../utils/assert-trailing-slash';
import { baseBannerStyle, offlineBannerStyle, offlineBannerContainerStyle } from './styles/bannerItemStyle';

const OfflineBanner = ({ linkUrl = 'https://mongodb.com/docs/', template = 'document' }) => {
  return (
    <div className={cx(offlineBannerContainerStyle({ template }))}>
      <LeafyBanner className={cx(baseBannerStyle, offlineBannerStyle({ template }))} variant={'warning'}>
        You are viewing an offline version of MongoDB documentation. Some page features might be unavailable. To view
        the latest version of the page or use interactive features, visit the&nbsp;
        <Link to={assertTrailingSlash(linkUrl)}>live page.</Link>
      </LeafyBanner>
    </div>
  );
};

OfflineBanner.propTypes = {
  linkUrl: PropTypes.string,
};

export default OfflineBanner;
