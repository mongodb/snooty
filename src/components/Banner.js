import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import LeafyBanner, { Variant as LeafyVariant } from '@leafygreen-ui/banner';
import ComponentFactory from './ComponentFactory';

export const alertMap = {
  info: LeafyVariant.Info,
  warning: LeafyVariant.Warning,
  danger: LeafyVariant.Danger,
  success: LeafyVariant.Success,
};

const StyledBanner = styled((props) => <LeafyBanner {...props} />)`
  /* Add margins below all child elements in the banner */
  & > div > div > * {
    margin: 0 0 12px;
  }

  & > div > div > *:last-child {
    margin: 0;
  }

  /* Remove margins on individual paragraphs */
  p {
    margin: 0;
  }
`;

const Banner = ({ nodeData: { children, options }, ...rest }) => {
  return (
    <StyledBanner variant={alertMap[options?.variant] || LeafyVariant.Info}>
      {children.map((child, i) => (
        <ComponentFactory {...rest} key={i} nodeData={child} />
      ))}
    </StyledBanner>
  );
};

Banner.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
    variant: PropTypes.string.isRequired,
  }).isRequired,
};

export default Banner;
