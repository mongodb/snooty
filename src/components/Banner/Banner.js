import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import LeafyBanner, { Variant as LeafyVariant } from '@leafygreen-ui/banner';
import ComponentFactory from '../ComponentFactory';
import { baseBannerStyle } from './styles/bannerItemStyle';

export const alertMap = {
  info: LeafyVariant.Info,
  warning: LeafyVariant.Warning,
  danger: LeafyVariant.Danger,
  success: LeafyVariant.Success,
};

const StyledBanner = styled((props) => <LeafyBanner {...props} />)`
  ${baseBannerStyle}
`;

const Banner = ({ nodeData: { children, options }, ...rest }) => {
  return (
    <StyledBanner variant={alertMap[options?.variant] || LeafyVariant.Info}>
      {children.map((child, i) => (
        <ComponentFactory {...rest} key={i} nodeData={child} />
      ))}
    </StyledBanner>
  );
};

Banner.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
    variant: PropTypes.string.isRequired,
  }).isRequired,
};

export default Banner;
