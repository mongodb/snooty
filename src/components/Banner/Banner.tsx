import React from 'react';
import { palette } from '@leafygreen-ui/palette';
import LeafyBanner, { Variant as LeafyVariant } from '@leafygreen-ui/banner';
import { css, cx } from '@leafygreen-ui/emotion';
import { getCurrLocale } from '../../utils/locale';
import ComponentFactory from '../ComponentFactory';
import type { BannerNode } from '../../types/ast';
import { baseBannerStyle } from './styles/bannerItemStyle';

export const alertMap = {
  info: LeafyVariant.Info,
  warning: LeafyVariant.Warning,
  danger: LeafyVariant.Danger,
};

const styleMapLight = {
  info: {
    backgroundColor: palette.blue.light3,
    color: palette.blue.dark2,
    borderColor: palette.blue.light2,
    linkColor: palette.blue.dark3,
    beforeColor: palette.blue.base,
    iconColor: palette.blue.base,
  },
  warning: {
    backgroundColor: palette.yellow.light3,
    color: palette.yellow.dark2,
    borderColor: palette.yellow.light2,
    linkColor: palette.yellow.dark3,
    beforeColor: palette.yellow.base,
    iconColor: palette.yellow.dark2,
  },
  danger: {
    backgroundColor: palette.red.light3,
    color: palette.red.dark2,
    borderColor: palette.red.light2,
    linkColor: palette.red.dark3,
    beforeColor: palette.red.base,
    iconColor: palette.red.base,
  },
};
const styleMapDark = {
  info: {
    backgroundColor: palette.blue.dark3,
    color: palette.blue.light2,
    borderColor: palette.blue.dark2,
    linkColor: palette.blue.light3,
    beforeColor: palette.blue.light1,
    iconColor: palette.blue.light1,
  },
  warning: {
    backgroundColor: palette.yellow.dark3,
    color: palette.yellow.light2,
    borderColor: palette.yellow.dark2,
    linkColor: palette.yellow.light3,
    beforeColor: palette.yellow.dark2,
    iconColor: palette.yellow.base,
  },
  danger: {
    backgroundColor: palette.red.dark3,
    color: palette.red.light2,
    borderColor: palette.red.dark2,
    linkColor: palette.red.light3,
    beforeColor: palette.red.base,
    iconColor: palette.red.light1,
  },
};

const bannerStyle = ({ variant }: BannerNode['options']) => css`
  ${baseBannerStyle}
  background-color: ${styleMapLight[variant].backgroundColor};
  color: ${styleMapLight[variant].color};
  border-color: ${styleMapLight[variant].borderColor};
  // copied from LG
  ::before {
    background: linear-gradient(to left, transparent 6px, ${styleMapLight[variant].beforeColor} 6px);
  }
  a {
    color: ${styleMapLight[variant].linkColor};
    :hover {
      color: ${styleMapLight[variant].color};
      text-decoration-color: ${styleMapLight[variant].color};
    }
  }
  > svg {
    color: ${styleMapLight[variant].iconColor};
  }

  .dark-theme & {
    background-color: ${styleMapDark[variant].backgroundColor};
    color: ${styleMapDark[variant].color};
    border-color: ${styleMapDark[variant].borderColor};
    ::before {
      background: linear-gradient(to left, transparent 6px, ${styleMapDark[variant].beforeColor} 6px);
    }
    a {
      color: ${styleMapDark[variant].linkColor};
      :hover {
        color: ${styleMapDark[variant].color};
        text-decoration-color: ${styleMapDark[variant].color};
      }
    }
    > svg {
      color: ${styleMapDark[variant].iconColor};
    }
  }
`;

interface BannerProps {
  nodeData: BannerNode;
}

const Banner = (props: BannerProps) => {
  // handling the nested destructing in the body, to allow cleaner debugging.
  const { nodeData, ...rest } = props;
  const { children, options } = nodeData;
  // Get the current locale (language + region) i.e. es-US, fr-FR
  const locale = getCurrLocale();

  // if banner has option locale, then only render the banner for said translated page.
  // Let us first do a string to array conversion for better accuracy
  const locales = options?.locale?.split(',');
  if (locales && !locales.includes(locale)) {
    return <div />;
  }

  return (
    <LeafyBanner className={cx(bannerStyle({ variant: alertMap[options?.variant] || LeafyVariant.Info }))}>
      {children.map((child, i) => (
        <ComponentFactory {...rest} key={i} nodeData={child} />
      ))}
    </LeafyBanner>
  );
};

export default Banner;
