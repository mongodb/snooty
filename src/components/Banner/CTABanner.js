import React from 'react';
import PropTypes from 'prop-types';
import { css } from '@emotion/core';
import { uiColors } from '@leafygreen-ui/palette';
import Icon, { glyphs } from '@leafygreen-ui/icon';
import ComponentFactory from '../ComponentFactory';
import { baseBannerStyle } from './styles/bannerItemStyle';

const videoBannerStyling = css`
  ${baseBannerStyle};
  background-color: ${uiColors.blue.light3};
  border: 1px solid ${uiColors.blue.light2};
  border-radius: 6px;
  display: flex;
  position: relative;
  font-size: 14px;
  padding: 9px 12px 9px 20px;
  color: ${uiColors.blue.dark2};
  margin: 24px 0px;

  > p {
    margin-left: 35px;
  }
`;

const lgIconStyling = css`
  width: 26px;
  height: 26px;
  background-color: ${uiColors.blue.light2};
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid ${uiColors.blue.base};
  border-radius: 20px;
  margin-left: -5px;
`;

const linkStyling = css`
  text-decoration: none !important;
`;

const CTABanner = ({ nodeData: { children, options }, ...rest }) => {
  // Handles case sensitivity for specified icons
  let lgIcon = 'Play';
  if (options?.icon) {
    const standardizeCaseLGIcon = options.icon.charAt(0).toUpperCase() + options.icon.slice(1).toLowerCase();
    if (standardizeCaseLGIcon in glyphs) {
      lgIcon = standardizeCaseLGIcon;
    }
  }

  return (
    <a href={options?.url} css={linkStyling}>
      <div css={videoBannerStyling}>
        <div css={lgIconStyling}>
          <Icon glyph={lgIcon} fill={uiColors.blue.base} />
        </div>
        {children.map((child, i) => (
          <ComponentFactory {...rest} key={i} nodeData={child} />
        ))}
      </div>
    </a>
  );
};

CTABanner.propTypes = {
  nodeData: PropTypes.shape({
    children: PropTypes.arrayOf(PropTypes.object).isRequired,
    options: PropTypes.shape({
      url: PropTypes.string.isRequired,
      icon: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default CTABanner;
