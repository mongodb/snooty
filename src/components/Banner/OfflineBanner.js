import React from 'react';
import PropTypes from 'prop-types';
import LeafyBanner from '@leafygreen-ui/banner';
import { cx } from '@leafygreen-ui/emotion';
import Link from '../Link';
import { baseBannerStyle, offlineBannerStyle, OfflineBannerContainer } from './styles/bannerItemStyle';

const OfflineBanner = ({ linkUrl = 'https://mongodb.com/docs', template = 'document' }) => {
  return (
    <OfflineBannerContainer template={template}>
      <LeafyBanner className={cx(baseBannerStyle, offlineBannerStyle)} variant={'warning'}>
        You are viewing an offline version of MongoDB documentation. Some page features might be unavailable. To view
        the latest version of the page or use interactive features, visit the&nbsp;
        <Link to={linkUrl}>live page.</Link>
      </LeafyBanner>
    </OfflineBannerContainer>
  );
};

OfflineBanner.propTypes = {
  linkUrl: PropTypes.string,
};

export default OfflineBanner;
