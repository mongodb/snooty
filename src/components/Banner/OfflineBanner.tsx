import React from 'react';
import PropTypes from 'prop-types';
import LeafyBanner from '@leafygreen-ui/banner';
import { cx } from '@leafygreen-ui/emotion';
import Link from '../Link';
import { assertTrailingSlash } from '../../utils/assert-trailing-slash';
import type { BannerTemplateType } from './styles/bannerItemStyle';
import { baseBannerStyle, offlineBannerStyle, offlineBannerContainerStyle } from './styles/bannerItemStyle';

interface OfflineBannerProps {
  linkUrl?: string;
  template?: BannerTemplateType;
}

const OfflineBanner = ({ linkUrl = 'https://mongodb.com/docs/', template = 'document' }: OfflineBannerProps) => {
  return (
    <div className={cx(offlineBannerContainerStyle({ template }))}>
      <LeafyBanner className={cx(baseBannerStyle, offlineBannerStyle({ template }))} variant={'warning'}>
        You are viewing an offline version of MongoDB documentation. Some page features might be unavailable. To view
        the latest version of the page or use interactive features, visit the&nbsp;
        <Link to={assertTrailingSlash(linkUrl)}>live page.</Link>
      </LeafyBanner>
    </div>
  );
};

OfflineBanner.propTypes = {
  linkUrl: PropTypes.string,
};

export default OfflineBanner;
