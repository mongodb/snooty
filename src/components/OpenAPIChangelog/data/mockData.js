export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
