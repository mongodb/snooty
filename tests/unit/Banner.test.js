import React from 'react';
import { shallow } from 'enzyme';
import Banner from '../../src/components/Banner';

// data for this component
import mockData from './data/Banner.test.json';

it('renders a Banner correctly', () => {
  const tree = shallow(<Banner nodeData={mockData} />);
  expect(tree).toMatchSnapshot();
});
