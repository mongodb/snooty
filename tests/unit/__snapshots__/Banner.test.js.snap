// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Banner correctly 1`] = `
<StyledBanner
  variant="info"
>
  <ComponentFactory
    key="0"
    nodeData={
      Object {
        "position": Object {
          "start": Object {
            "line": 6,
          },
        },
        "type": "text",
        "value": "These instructions are for installing MongoDB directly from",
      }
    }
  />
</StyledBanner>
`;
